[15.0][ADD] product_attribute_description

dsolanki-initos opened the pull request:
Depends on https://github.com/OCA/website/pull/907

Commit:
[ADD] product_sequence: Allows to associate a sequence to the product reference. The reference (default code) is unique (SQL constraint) and required.

sebastienbeau committed:
[MERGE] merge translation commited by launchpad on extra-addons6.0

bguillot committed:
[FIX] fix write all product with default_code together and then write each product without default_code

eLBati committed:
[FIX] resetting sequence code when updating module

Commit:
[REM] use Model instead of osv.osv

some minor reformatting in the process

sebastienbeau committed:
[FIX] product_sequence : write should support list or integer

Commit:
[MIGR] migration of product_sequence to openerp 7.0 (required for product_images)

Commit:
FIX description of module product_custom_attributes.
Version numbers should only have 2 digits.

Commit:
Launchpad automatic translations update.

Commit:
Set all addons to uninstallable and move them into __unported__ folder

Commit:
MIG product_sequence

Commit:
Added german translation

yvaucher committed:
Add bug tracker link on README.rst

gdgellatly committed:
product_sequence

Add pre_init_hook to manually set nulls and '/' default codes
 to a unique code to greenify branch
add pragma to pre_init method as it must have completed successfully
if module installs

yvaucher committed:
Add missing default oca icons

SodexisTeam committed:
product_sequence (#156)

atchuthan committed:
[MIG][10.0] migrate product_sequence

oca-transbot committed:
OCA Transbot updated translations from Transifex

LoisRForgeFlow committed:
[11.0][MIG/IMP] product_sequence:
* Migration to v11
* Support sequence by Product Category

JordiBForgeFlow committed:
fixes and adds tests

sbidoul committed:
remove obsolete .pot files [ci skip]

oca-travis committed:
[UPD] Update product_sequence.pot

rousseldenis committed:
[10.0][ADD] product_assortment

sergio-teruel committed:
[11.0][NEW] product_secondary_unit: New module for setting a secondary unit for a product (#377)

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update product_secondary_unit.pot

sergio-teruel committed:
[11.0][ADD] product_secondary_unit: Add default secondary unit for purchases

Commit:
Added translation using Weblate (French)

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (17 of 17 strings)

Translation: product-attribute-11.0/product-attribute-11.0-product_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-11-0/product-attribute-11-0-product_secondary_unit/fr/

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

oca-travis committed:
[UPD] Update product_secondary_unit.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

sudhir-erpharbor committed:
[MIG] Migrated product_sequence to v12

mvrodriguez committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (16 of 16 strings)

Translation: product-attribute-11.0/product-attribute-11.0-product_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-11-0/product-attribute-11-0-product_secondary_unit/es/

sergio-teruel committed:
[11.0][IMP] product_secondary_unit: Make secondary unit name translatable

kittiu committed:
[12.0][MIG] product_secondary_unit

oca-travis committed:
[UPD] Update product_secondary_unit.pot

tbaden committed:
[FIX][10.0] product_assortment travis warning

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-12.0/product-attribute-12.0-product_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-12-0/product-attribute-12-0-product_secondary_unit/

rousseldenis committed:
[12.0][MIG] product_assortment

OCA-git-bot committed:
[UPD] README.rst

Tardo committed:
[IMP] product_sequence: Suggested PR changes

oca-travis committed:
[UPD] Update product_sequence.pot

Commit:
[IMP] product_assortment: black, isort

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-12.0/product-attribute-12.0-product_sequence
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-12-0/product-attribute-12-0-product_sequence/

digitalsatori committed:
Added translation using Weblate (Chinese (Simplified))

digitalsatori committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (16 of 16 strings)

Translation: product-attribute-12.0/product-attribute-12.0-product_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-12-0/product-attribute-12-0-product_secondary_unit/zh_CN/

Commit:
[MIG] product_assortment: Migration to 13.0

ps-tubtim committed:
[IMP] product_secondary_unit: black, isort

ps-tubtim committed:
[MIG] product_secondary_unit: Migration to 13.0

sergio-teruel committed:
[12.0][FIX] product_sequence: Keep default_code when copy products if dictionary values contains the default_code key.

OCA-git-bot committed:
product_sequence 12.0.1.0.1

oca-travis committed:
[UPD] Update product_secondary_unit.pot

sergio-teruel committed:
[IMP] product_sequence: black, isort

sergio-teruel committed:
[MIG] product_sequence: Migration to v13.0

oca-travis committed:
[UPD] Update product_sequence.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-13.0/product-attribute-13.0-product_sequence
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_sequence/

benbrich committed:
Translated using Weblate (German)

Currently translated at 100.0% (9 of 9 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_sequence
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_sequence/de/

sergio-teruel committed:
[IMP] product_secondary_unit: Remove product secondary units if a product is deleted

alan196 committed:
Added translation using Weblate (Spanish (Mexico))

alan196 committed:
Translated using Weblate (Spanish (Mexico))

Currently translated at 100.0% (16 of 16 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_secondary_unit/es_MX/

OCA-git-bot committed:
product_secondary_unit 13.0.1.1.0

OCA-git-bot committed:
pre-commit update

OCA-git-bot committed:
pre-commit update

didierdonze committed:
Translated using Weblate (French)

Currently translated at 100.0% (16 of 16 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_secondary_unit/fr/

didierdonze committed:
Translated using Weblate (French)

Currently translated at 100.0% (9 of 9 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_sequence
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_sequence/fr/

benwillig committed:
[FIX] avoid errors when trying to open assortment products

MiquelRForgeFlow committed:
[UPD] Eficent -> ForgeFlow

OCA-git-bot committed:
[UPD] README.rst

sbidoul committed:
[FIX] product_assortment: development_status

alan196 committed:
[MIG] product_secondary_unit: Migration to 14.0

oca-travis committed:
[UPD] Update product_secondary_unit.pot

watthanun committed:
[MIG] product_sequence: Migration to 14.0

oca-travis committed:
[UPD] Update product_sequence.pot

OCA-git-bot committed:
[UPD] README.rst

CarlosRoca13 committed:
[IMP] product_assortment: added partners to products assortments

sergio-teruel committed:
[IMP] product_secondary_unit: Add mixin model to be inherited from other models.
TT27363

pedrobaeza committed:
[FIX] product_assortment: Avoid errors on invalid models

You can have an ir.filters entry with an invalid model. That shouldn't
break system in case of accessing the filter.

sergio-teruel committed:
[FIX] product_secondary_unit: Avoid to recompute secondary_uom_qty when secondary uom changes

[FIX] product_secondary_unit: Avoid cache missed

khaeusler committed:
Use parent categories to determine the prefix if the category has no settings for the prefix.

oca-travis committed:
[UPD] Update product_sequence.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
product_sequence 14.0.2.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-14.0/product-attribute-14.0-product_sequence
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_sequence/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (18 of 18 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_sequence
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_sequence/nl/

claudiagn committed:
Added translation using Weblate (Catalan)

claudiagn committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (20 of 20 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_assortment/ca/

xavier-bouquiaux committed:
[14.0][MIG] product_assortment

rousseldenis committed:
[14.0][ADD] product_category_code

rousseldenis committed:
[14.0][ADD] product_category_code_unique

rousseldenis committed:
[14.0][IMP] product_category_code_unique: Add sequence

rousseldenis committed:
[14.0][IMP] product_category_code: Add code in list view

mourad-ehm committed:
[IMP] add dependency type for product_secondary_unit_mixin

rousseldenis committed:
[14.0][IMP] product_assortment: Fallback to create() for model_id

oca-travis committed:
[UPD] Update product_category_code.pot

mourad-ehm committed:
Apply suggestions from code review

Co-authored-by: Hpar <raphael.reverdy@akretion.com>

Kev-Roche committed:
fix ir_filters

mourad-ehm committed:
[FIX] refactor code

sergio-teruel committed:
[IMP] product_assortment: Allow to use assortment based on partner domain
TT30584

Kev-Roche committed:
[FIX] formatting and minor changes

Kev-Roche committed:
rework and test dependency_type

oca-travis committed:
[UPD] Update product_secondary_unit.pot

OCA-git-bot committed:
product_secondary_unit 14.0.1.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-14.0/product-attribute-14.0-product_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_secondary_unit/

legalsylvain committed:
[ADD][12.0] new module product_net_weight.
- Add new net_weight field on product models;
- Rename weight field with 'Gross Weight' string;

CarlosRoca13 committed:
[IMP] product_assortment: add icon to menu

OCA-git-bot committed:
[ADD] icon.png

Kev-Roche committed:
[14.0][FIX] secondary_uom: independent behavior

OCA-git-bot committed:
product_secondary_unit 14.0.1.0.2

pedrobaeza committed:
[ADD] product_pricelist_by_contact: Pricelist per contact

This module allows to set a different pricelist per each contact, so that such
pricelist is used in the sales flow.

Usage
=====

* Go to *Contacts* (you need to install module `contacts` for having this menu entry).
* Select one contact that is not the parent one.
* On the tab "Sales & Purchases", you will be able to select a different pricelist
  from its parent contact.
* If you change the pricelist in the parent contact, the pricelist of the children
  contacts will be overwritten with this one, no matter the previous value.

Known issues / Roadmap
======================

* Not requiring to override any sales behavior is possible due to the way Odoo
  retrieves the pricelist directly associated to the contact, but no code is put
  here for assuring that such behavior is preserved across time.

Commit:
[FIX] remove useless depends on domain leading to orm crash

OriolVForgeFlow committed:
[13.0][ADD] product_variant_attribute_name_manager

hailangvn committed:
[FIX] product_assortment: use safe_eval.datetime

oca-travis committed:
[UPD] Update product_variant_attribute_name_manager.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

remihb committed:
[MIG] product_secondary_unit: Migration to 15.0

OriolVForgeFlow committed:
[14.0][MIG] product_variant_attribute_name_manager: Migration to 14.0

GuillemCForgeFlow committed:
[13.0][ADD]product_supplierinfo_archive

manuelregidor committed:
[IMP]product_net_weight: black, isort, prettier

manuelregidor committed:
[MIG]product_net_weight: Migration to 14.0

oca-travis committed:
[UPD] Update product_variant_attribute_name_manager.pot

OCA-git-bot committed:
[UPD] README.rst

OriolVForgeFlow committed:
[IMP] product_variant_attribute_name_manager: black, isort, prettier

OriolVForgeFlow committed:
[15.0][MIG] product_variant_attribute_name_manager: Migration to 15.0

oca-travis committed:
[UPD] Update product_supplierinfo_archive.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

sebalix committed:
[FIX] Fix access right issues when using actions

GuillemCForgeFlow committed:
[13.0][FIX]product_supplierinfo_archive: Added field on view and changed filters

rousseldenis committed:
[14.0][IMP] product_category_code_unique: Add False company_id data

Rad0van committed:
[IMP] product_category_code: black, isort, prettier

Rad0van committed:
[MIG] product_category_code: Migration to 15.0

BertVGroenendael committed:
[IMP] product_sequence: black, isort, prettier

BertVGroenendael committed:
[MIG] product_sequence: Migration to 15.0

Noel000 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (24 of 24 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_assortment/ca/

oca-travis committed:
[UPD] Update product_category_code_unique.pot

etobella committed:
[IMP] product_supplier_info_for_customer: Promote to stable

SicurSam committed:
Added translation using Weblate (Italian)

SicurSam committed:
Translated using Weblate (Italian)

Currently translated at 70.8% (17 of 24 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_assortment/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (24 of 24 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_assortment/it/

SicurSam committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (24 of 24 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_assortment/it/

dsolanki-initos committed:
[15.0][ADD] product_attribute_description

dsolanki-initos committed:
[15.0][ADD] product_attribute_description

fshah-initos reviewed (approved):
Code review and functional tests.
Looks good to me.

rousseldenis commented:
@dsolanki-initos Is it a real migration or a new module ?

jans23 commented:
Answering on behalf: That's a new module.

rousseldenis commented on product_attribute_description/__manifest__.py:
```diff
@@ -0,0 +1,27 @@
+    This module adds the information icon next to the attribute name in the
+    website and display the related description on mouse hover.
+     """,
+    "depends": ["website_sale"],
```
Why depending on this ???

jans23 commented on product_attribute_description/__manifest__.py:
```diff
@@ -0,0 +1,27 @@
+    "name": "Product Attribute Description",
+    "version": "15.0.1.0.0",
+    "category": "Website",
+    "author": "ERP Harbor Consulting Services,"
```
Add Nitrokey GmbH too.

dsolanki-initos replied:
Hello @jans23,

I've already added and you can check this URL for the same 

https://github.com/OCA/product-attribute/blob/014744062386d7c00950a59081fcff443adcb407/product_attribute_description/__manifest__.py#L5-L8

Regards,

Dhara Solanki

CarlosRoca13 committed:
[IMP] product_assortment: black, isort, prettier

CarlosRoca13 committed:
[MIG] product_assortment: Migration to 15.0

sergio-teruel committed:
[IMP] product_assortment: Add blacklist product domain to exclude some products
TT36814

flachica committed:
[IMP] product_net_weight: black, isort

flachica committed:
[MIG][15.0] Standard migration product_net_weight

Commit:
[UPD] Update product_secondary_unit.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

Commit:
[UPD] Update product_net_weight.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
product_net_weight 15.0.1.0.1

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] addons table in README.md

Rad0van commented:
If it's a new module then the PR name is misleading. If it's aimed at e-commerce it should be in that repo. Or probably even better: split into 2 modules: one adding the description in this repo and the other one showing it in e-commerce. See for example https://github.com/OCA/brand/tree/14.0/product_brand and https://github.com/OCA/e-commerce/tree/14.0/website_sale_product_brand

Commit:
[UPD] Update product_category_code.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

rolandojduartem committed:
[MIG] product_category_code_unique: Migrate to v15.0.

Commit:
[UPD] Update product_category_code_unique.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
product_supplierinfo_for_customer 15.0.1.0.1

OCA-git-bot committed:
[UPD] addons table in README.md

dsolanki-initos commented:
> If it's a new module then the PR name is misleading. If it's aimed at e-commerce it should be in that repo. Or probably even better: split into 2 modules: one adding the description in this repo and the other one showing it in e-commerce. See for example https://github.com/OCA/brand/tree/14.0/product_brand and https://github.com/OCA/e-commerce/tree/14.0/website_sale_product_brand

Okay I will split it

Commit:
[UPD] Update product_code_mandatory.pot

Commit:
[UPD] Update product_sequence.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-15.0/product-attribute-15.0-product_code_mandatory
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-15-0/product-attribute-15-0-product_code_mandatory/

dsolanki-initos commented:
> If it's a new module then the PR name is misleading. If it's aimed at e-commerce it should be in that repo. Or probably even better: split into 2 modules: one adding the description in this repo and the other one showing it in e-commerce. See for example https://github.com/OCA/brand/tree/14.0/product_brand and https://github.com/OCA/e-commerce/tree/14.0/website_sale_product_brand

FYI, I've split it into two modules here is another PR https://github.com/OCA/website/pull/907

cesar-tecnativa committed:
[IMP] product_pricelist_by_contact: black, isort, prettier

cesar-tecnativa committed:
[MIG] product_pricelist_by_contact: Migration to 15.0

Commit:
[UPD] Update product_pricelist_by_contact.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

JasminSForgeFlow committed:
[MIG] product_supplierinfo_archive: Migration to 14.0

JasminSForgeFlow committed:
[IMP] product_supplierinfo_archive: black, isort, prettier

JasminSForgeFlow committed:
[MIG] product_supplierinfo_archive: Migration to 15.0

dsolanki-initos commented:
@CarlosRoca13 Can you review this PR?

dsolanki-initos commented:
@chienandalu Can you please review this PR?

dsolanki-initos commented:
@olgamarcocb Can you please review this PR?

ikapasi-initos reviewed (approved):
Code review and functional test.
LGTM

Commit:
[UPD] Update product_assortment.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-15.0/product-attribute-15.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-15-0/product-attribute-15-0-product_assortment/

jacekmichalski committed:
Translated using Weblate (Polish)

Currently translated at 100.0% (5 of 5 strings)

Translation: product-attribute-15.0/product-attribute-15.0-product_manufacturer
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-15-0/product-attribute-15-0-product_manufacturer/pl/

dsolanki-initos commented:
@ernestotejeda Could you please review this PR?

Commit:
[UPD] Update product_variant_attribute_name_manager.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

Commit:
[UPD] Update product_supplierinfo_for_customer.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-15.0/product-attribute-15.0-product_supplierinfo_for_customer
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-15-0/product-attribute-15-0-product_supplierinfo_for_customer/

dsolanki-initos commented:
@JordiBForgeFlow Can we merge this PR?

dsolanki-initos committed:
Merge branch '15.0-mig-product_attribute_description' of https://github.com/Nitrokey/odoo-product-attribute into 15.0-mig-product_attribute_description

jans23 commented:
@rousseldenis We did two internal reviews and incorporated yours and other reviewers feedback. Could you reopen this please?

dsolanki-initos commented:
Created new PR: https://github.com/OCA/product-attribute/pull/1255
